perun-core/src/main/java/cz/metacentrum/perun/core/api/AttributesManager.java

```diff
@@ -192,6 +192,27 @@
 	List<Attribute> getAttributes(PerunSession sess, Resource resource, Member member, boolean workWithUserAttributes) throws PrivilegeException, ResourceNotExistsException, InternalErrorException, MemberNotExistsException, WrongAttributeAssignmentException;
 
 	/**
+	 * Gets selected <b>non-empty</b> attributes associated with the member and the resource.
+	 * It returns member and member-resource attributes and also user and user-facility attributes if
+	 * workWithUserAttributes is true.
+	 * Attributes are selected by list of attr_names. Empty list means all attributes.
+	 *
+	 * @param sess perun session
+	 * @param resource to get the attributes from
+	 * @param member to get the attributes from
+	 * @param attrNames list of attributes to get
+	 * @param workWithUserAttributes if true returns also user-facility and member-resource attributes (user is automatically get from member a facility is get from resource)
```

Comment:
Doesn't match javadoc above:

workWithUserAttributes=false - member, member-resource
workWithUserAttributes=true - member, user, member-resource, user-facility

vs.

workWithUserAttributes=false - member, user
workWithUserAttributes=true - member-resource, user-facility

Basically, Bl version of javadoc is correct.